.changeset/grey-triangles-shout.md

```diff
@@ -0,0 +1,68 @@
+---
+"@fluidframework/tree": minor
+---
+---
+section: tree
+---
+
+Allow associating metadata with Node Schema
+
+Users of TreeView can now specify metadata when creating Node Schema.
+This includes system-understood metadata, e.g., `description`.
+
+Example:
+
+```typescript
+
+class Point extends schemaFactory.object("Point", {
+	x: schemaFactory.required(schemaFactory.number),
+	y: schemaFactory.required(schemaFactory.number),
+},
+{
+	metadata: { description: "A point in 2D space" }
+}) {}
+
+```
+
+Functionality like the experimental conversion of Tree Schema to [JSON Schema](https://json-schema.org/) (`getJsonSchema`) leverages such system-understood metadata to generate useful information.
```

Comment:
Are there some docs or release notes for this feature we could link to?

Reply:
We should introduce some generalized guidance around this.
 
There are a few things I could link to here:
1. The API docs on the website
  - Pros: Theoretically points to the most relevant information
  - Cons: Somewhat unstable (if the API is renamed, etc.)
2. The release notes containing the change that introduced the functionality
  - Pros: Stable
  - Cons: While the link is stable, the linked contents become less useful as updates to the APIs are made
3. The code file on GitHub
  - Pros: Honestly, I don't see any. Just included this as an option for completeness.
  - Cons: Unstable

Personally, I think option 1 is probably the most _useful_, but option 2 will be more stable. Option 3 is neither, IMO, just added for "completeness".

Reply:
```suggestion
Functionality like the experimental conversion of Tree Schema to [JSON Schema](https://json-schema.org/) ([getJsonSchema](https://github.com/microsoft/FluidFramework/releases/tag/client_v2.4.0#user-content-metadata-can-now-be-associated-with-field-schema-22564)) leverages such system-understood metadata to generate useful information.
```